src/connector/src/sink/remote.rs

```diff
@@ -0,0 +1,574 @@
+                .ok_or_else(|| SinkError::Remote("response stream closed unexpectedly".to_string()))
+        };
+    }
+}
+
+#[derive(Debug)]
+pub enum ConnectorClientImpl {
+    Grpc(ConnectorServiceClient<Channel>),
+    Mock,
+}
+
+#[derive(Debug)]
+pub struct RemoteSink {
+    pub sink_type: String,
+    properties: HashMap<String, String>,
+    epoch: Option<u64>,
+    batch_id: u64,
+    schema: Schema,
+    client: ConnectorClientImpl,
+    pub request_sender: Option<UnboundedSender<SinkTask>>,
+    pub response_stream: RemoteSinkResponseImpl,
+}
+
+pub struct RemoteSinkParams {
+    pub connector_addr: String,
+}
+
+impl RemoteSink {
+    pub async fn new(
+        config: RemoteConfig,
+        schema: Schema,
+        pk_indices: Vec<usize>,
+        sink_params: RemoteSinkParams,
+    ) -> Result<Self> {
+        let channel = Endpoint::from_shared(format!("http://{}", sink_params.connector_addr))
+            .map_err(|e| SinkError::Remote(format!("failed to connect channel: {:?}", e)))?
+            .initial_connection_window_size(MAX_CONNECTION_WINDOW_SIZE)
+            .initial_stream_window_size(STREAM_WINDOW_SIZE)
+            .tcp_nodelay(true)
+            .connect_timeout(Duration::from_secs(5))
+            .connect()
+            .await
+            .map_err(|e| SinkError::Remote(format!("failed to connect channel: {:?}", e)))?; // create client and start sink
+        let mut client = ConnectorServiceClient::new(channel);
+
+        let (request_sender, request_receiver) = mpsc::unbounded_channel::<SinkTask>();
+        request_sender
+            .send(SinkTask {
+                request: Some(SinkRequest::Start(StartSink {
+                    sink_config: Some(SinkConfig {
+                        sink_type: config.sink_type.clone(),
+                        properties: config.properties.clone(),
+                        table_schema: Some(TableSchema {
+                            columns: schema
+                                .fields()
+                                .iter()
+                                .map(|c| Column {
+                                    name: c.name.clone(),
+                                    data_type: parse_data_type(c.data_type.clone()) as i32,
+                                })
+                                .collect(),
+                            pk_indices: pk_indices.iter().map(|i| *i as u32).collect(),
+                        }),
+                    }),
+                })),
+            })
+            .map_err(|e| SinkError::Remote(e.to_string()))?;
+
+        let mut response = tokio::time::timeout(
+            Duration::from_secs(3),
+            client.sink_stream(Request::new(UnboundedReceiverStream::new(request_receiver))),
+        )
+        .await
+        .map_err(|e| SinkError::Remote(format!("failed to start sink: {:?}", e)))?
+        .map_err(|e| SinkError::Remote(format!("{:?}", e)))?
+        .into_inner();
+        let _ = response.next();
```

Comment:
May add a comment here why we call `next`. Besides, the next may also fail. We should check the result.

Reply:
We do need the comment, but I think unwrapping the awaited result would do the job.